### DISCUSSION TITLE
隐语v0.7.18版本更新，构建跨机构分布式底座RayFed

### DISCUSSION THREAD

**Discussion by halosecret00** [secretflow/secretflow#261] [General]
![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI346ic5nmRKwFKz0PdiaHE2MwdkRQ2BXO3E4Lo93UUc17vvKibvyC7S4lcykpWibt4EMz7bn3SlzMdkuAw/640?wx_fmt=png)

本次隐语v0.7.18版本更新，带来了全新的功能优化和体验。在多方安全计算算法层，明密文调度层，基础设备层，文档等方面都做了优化，并发布全新的跨机构联邦分布式底座（RayFed），提供跨机构的分布式联邦执行能力，在保持隐语集中式编程易用性的基础上，提升整体系统的安全性。

![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI34r5GjvEpDRZPBHIKiaNW1AbY9eZXRUQre5VKLAQSrLABLqAFB39UicibalCkcXFv8RQBibKSSVbJszKQ/640?wx_fmt=png)

**🌟详细更新内容如下：**

**1、多方安全计算：**

*   互联互通：发布ECDH-PSI算法开放协议，开启通过开放算法协议实践，来实现和其他平台的互联互通；
    
*   隐私安全求交：支持基于ECDH非平衡PSI的python调用接口；
    
*   支持广义线性模型（GLM），该模型多用于保险场景。
    

**2、联邦学习：**

*   水平：提供支持稀疏压缩优化的明文sparse聚合器；
    
*   垂直：提供稀疏化的通信优化能力；SLModel支持复杂模型，支持basenet输出多路hiddens，便于灵活配置网络结构。
    

**3、基础设备及原语：**

*   与 Intel 合作在 HEU 中新增 IPCL 半同态加密算法，该算法支持 AVX512-IFMA 指令集和 Intel QAT 硬件加速
    

**4、文档：**

*   在隐语官网中集成了文档，提升国内访问速度，并提供全球加速。
    

**5、明密文调度框架：**

*   多控制器模式模式：基于ray构建了新的跨机构联邦分布式底座（RayFed），支持多控制器执行，提升生产模式的安全性。
    

另外，配合本次0.7.18版本更新，开源社区Meetup杭州站将于**2月中旬**与伙伴们线下见面！届时现场将对分布式调度引擎RayFed、算法协议互联互通等核心亮点，进行更加深度地介绍讲解，期待位广大开发者朋友提供更真切的感受体会，针对版本与社区进行面对面的反馈交流。【Meetup杭州站 预报名已开启】扫描下方二维码进行预报名，社区将收集反馈在现场针对解答！

![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI346ic5nmRKwFKz0PdiaHE2MwdDdz8Jp8DnJm5FYvSZmVibVuVBga1e3PtBmYhHW4IGS4TGBKic57XSJsQ/640?wx_fmt=png)

 联系我们：

公众号：隐语的小剧场

B站：隐语secretflow

邮箱：secretflow-contact@service.alipay.com

**Comment by lidh15** [comment]
It's nice to see the new release! I wonder how to do PSI, maybe ECDH, on rayfed. Currently, the demo is simply doing some addition, which is not that meaningful to real applications like PSI and PIR. And I failed to find documentation on this topic.

**Reply by icavan** [reply]
> It's nice to see the new release! I wonder how to do PSI, maybe ECDH, on rayfed. Currently, the demo is simply doing some addition, which is not that meaningful to real applications like PSI and PIR. And I failed to find documentation on this topic.

Here is the [PSI Tutorial](https://www.secretflow.org.cn/docs/secretflow/zh_CN/tutorial/PSI_On_SPU.html)

All features work under [rayfed](https://www.secretflow.org.cn/docs/secretflow/zh_CN/getting_started/deployment.html#production), including the PSI. As for PIR, we'll release it later. Stay tuned.
